src/OpenTelemetry/Trace/BatchingActivityProcessor.cs

```diff
@@ -338,7 +338,11 @@ private async Task<int> ExportBatchAsync(CancellationToken cancellationToken)
                     this.batch.Add(nextActivity);
                 }
 
+                var previous = Sdk.SuppressInstrumentation;
+                Sdk.SuppressInstrumentation = true;
                 var result = await this.exporter.ExportAsync(this.batch, cancellationToken).ConfigureAwait(false);
+                Sdk.SuppressInstrumentation = previous;
```

Comment:
Please consider a try-finally block (or adding the `using` syntactical sugar) to avoid accidental context leakage.

Reply:
Yep, good call. I do like the `using` approach and had planned on going that route.